### ISSUE TITLE
Documentation example problemo

### ISSUE THREAD

**Issue by jamesplease** [ember-animation/ember-animated#105]
It is possible to create unexpected behaviors in the "The power of promises" example on the website. If you click two options back and forth on a steady interval, before the animation completes, the animation gets into a strange state. As you continue to do this, it seems like the velocity of the items slowly increases.

A coworker of mine was also able to get the cards into a broken state, where cards would stop in a mid-transition state.

Clicking in this way could be written off as nonstandard user behavior, but I frequently require animations where the user can input before the transition completes, so this kind of use case is important to me. To give a concrete example, consider keyboard navigation with a 350ms input throttle paired with a 450ms animation duration.

Perhaps this is just a consequence of the specific way that this example was set up; I'm not sure! But I figured I'd open an issue anyway.

**Comment by willrax** [comment]
Here are the screenshots of the issue after clicking back and forth a little bit:

<img width="895" alt="Screen Shot 2019-05-08 at 8 57 16 AM" src="https://user-images.githubusercontent.com/94960/57390918-3faca400-7172-11e9-8129-024ff9bf92bf.png">
<img width="724" alt="Screen Shot 2019-05-08 at 8 57 41 AM" src="https://user-images.githubusercontent.com/94960/57390923-40453a80-7172-11e9-8a53-2b48a16d69b7.png">

**Comment by ef4** [comment]
Yes, making all interruptions work correctly is one of the founding reasons this library exist.

This is a bug in this particular demo, although I would like to change a default behavior to make this kind of bug harder to stumble into.

The `collapse`, `shuffle`, and `shuffleWithStagger` transitions all need to place the `sentSprites` into their final positions immediately as the transition begins:

```js
for (let sprite of sentSprites) {
  sprite.moveToFinalPosition();
}
```

That fixes the bug.

I would like to make this behavior the default, but it will require a corresponding change to the way motions are defined, because right now they assume this behavior.

**Comment by jamesplease** [comment]
Thanks for the reply!

> transitions all need to place the `sentSprites` into their final positions immediately as the transition begins

I don't fully understand the technical solution you described (as I'm not currently a user of the lib), but I have a follow up question: will items ever "jump" in the solution you described, or will they smoothly transition from the current animation into the new one?

**Comment by ef4** [comment]
In more detail: the `sentSprites` are present but invisible during all these transitions. That is, you always have two copies of each image: one in the sending component, and one in the receiving component. The demo always moves the receiving one and leaves the sending one invisible. 

In an effort to be helpful, ember-animated pre-positions both the received and sent images at the initial position (the place where the move should start). But since the demo doesn't do anything with the sent image, the sent image lingers there in that position. 

But when an interruption occurs, the sent image is still lingering in that weird position, and it can continue to influence the starting positions of the next transition.

The change I want to make is to not automatically move the sent sprites (and indeed, not automatically move any sprites), and instead require each motion to affirmatively declare when it wants a particular sprite to go to initial or final position whenever it expects one of those.

**Comment by ef4** [comment]
Nothing should ever jump. If you patch the code snippet I shared above into each of the three transition functions in that demo, everything is smooth on matter how many times you go back and forth clicking wildly.

Well, there is one exception in this demo, although it seems to be intention. The z-index changes take place instantly, so that whatever images are the incoming ones jump above the outgoing ones.

Probably there is a different strategy that would avoid this, at the cost of sometimes allowing outgoing images to be above incoming ones.

**Comment by jamesplease** [comment]
Very cool. I appreciate the thorough response, @ef4 ! I wasn’t able to change the example on the site, but it’s nbd, I believe ya’ 🙂 .

Since it sounds like the resolution to this problem is known, would you like me to close this issue?

**Comment by ef4** [comment]
This can close when #107 merges. Thanks.
